File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.py
class Solution(object):
    def maxFreqSum(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt = Counter(s)
        vowels = set('aeiou')
        
        max_vowel = 0
        max_consonant = 0
        
        for c, freq in cnt.items():
            if c in vowels:
                if freq > max_vowel:
                    max_vowel = freq
            else:
                if freq > max_consonant:
                    max_consonant = freq
                    
        return max_vowel + max_consonant

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {
    let map = {};
    for (let i=0; i<s.length; i++) {
        if (map[s[i]]){
            map[s[i]]++;
        } else {
            map[s[i]] = 1;
        }
    }

    let vowels = ['a', 'e', 'i', 'o', 'u'];
    let maxVowel = 0;
    let maxConsonant = 0;
    for(let i=0; i<s.length; i++) {
        if (vowels.includes(s[i])) {
            maxVowel = Math.max(maxVowel, map[s[i]]);
        } else {
            maxConsonant = Math.max(maxConsonant, map[s[i]]);
        }
    }

    return maxConsonant + maxVowel;
};